pageserver/src/context.rs

```diff
@@ -100,15 +100,24 @@ use crate::{
     task_mgr::TaskKind,
     tenant::Timeline,
 };
+use futures::FutureExt;
+use futures::future::BoxFuture;
+use std::future::Future;
+use tracing_utils::perf_span::{PerfInstrument, PerfSpan};
+
+use tracing::{Dispatch, Span};
 
 // The main structure of this module, see module-level comment.
+#[derive(Clone)]
 pub struct RequestContext {
```

Comment:
Would prefer if the public type was not clonable. Users should be forced to go through the RequestContext APIs to create child contexts.

Reply:
Done in https://github.com/neondatabase/neon/pull/11140/commits/6ce44f5ee50e29914838423b68c4c7e5c45d4ed2